lib/index.ts

```diff
@@ -87,19 +87,35 @@ export async function fetchBeaconByTime(client: ChainClient, time: number): Prom
 }
 
 // an async generator emitting beacons from the latest round onwards
-export async function* watch(client: ChainClient, abortController: AbortController): AsyncGenerator<RandomnessBeacon> {
-    while (!abortController.signal.aborted) {
-        const info = await client.chain().info()
-        const beacon = await client.latest()
-        yield validatedBeacon(client, beacon)
+export async function* watch(
+    client: ChainClient,
+    abortController: AbortController,
+    options: WatchOptions = defaultWatchOptions,
+): AsyncGenerator<RandomnessBeacon> {
+    const info = await client.chain().info()
+    let currentRound = roundAt(Date.now(), info)
 
-        const nextRoundTime = roundTime(info, beacon.round + 1)
+    while (!abortController.signal.aborted) {
         const now = Date.now()
+        await sleep(roundTime(info, currentRound) - now)
```

Comment:
Should we offset this by 10ms maybe to account for the network latency? That could help fetch randomness faster, maybe

Reply:
But we actually don't know the average latency at which our HTTP relays are serving randomness...